fix application crushes with error in AsyncSearch().count() method

jellex opened the pull request:
I tried using AsyncSearh with next code:
```python
from elasticsearch_dsl import AsyncSearch

search = AsyncSearch(using='my_connection_name', index='my_index_name')
search = search.filter('term', docType='my_docType_name')
result = search.count()
```
and I get the following error:
```python
  <...>
  File "<..>/elasticsearch_dsl/_async/search.py", line 430, in count
    return await es.count(index=self._index, body=d, **self._params)["count"]
TypeError: 'coroutine' object is not subscriptable
```


before this bug fix, when calling AsyncSearch().count(), the application crashes with the error `TypeError: 'coroutine' object is not subscriptable`

jellex committed:
fix application crushes with error in AsyncSearch().count() method

before this bug fix, when calling AsyncSearch().count(), the application crashes with the error "TypeError: 'coroutine' object is not subscriptable"

jellex committed:
fix lint errors

Andrew-Chen-Wang commented:
You can just do `return (await es.count(index=self._index, body=d, **self._params))["count"]` to make it in one line.

jellex committed:
Merge branch 'sethmlarson:async-io' into async-io

jellex committed:
delete devide for Data field

Not needed divide by a float to preserve milliseconds on the datetime, because 'data' already in the required format.
